recipe/meta.yaml

```diff
@@ -7,36 +10,89 @@ package:
 source:
   fn: torchvision-{{ version }}.tar.gz
   url: https://github.com/pytorch/vision/archive/v{{ version }}.tar.gz
-  sha256: 91f1e21121dca7e14ec35b5f2127da77c8c52e61d39957ab5dd4a719a837a04c
+  sha256: 9a866c3c8feb23b3221ce261e6153fc65a98ce9ceaa71ccad017016945c178bf
+  patches:
+    # skip failing test thats already marked as flaky upstream, see
+    # https://github.com/pytorch/vision/blob/v0.8.2/test/test_models.py#L90
+    - patches/0001-skip-flaky-test.patch
 
 build:
-  number: 1000
-  skip: True  # [not linux64]
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  number: {{ number }}
+  skip: true  # [not linux]
 
 requirements:
   build:
-    - python
-    - setuptools
-
-  run:
-    - python
-    - pillow >=4.1.1
-    - numpy >=1.11
-    - pytorch >=0.3
-    - six
-
-test:
-  imports:
-    - torchvision
-    - torchvision.datasets
-    - torchvision.models
-    - torchvision.transforms
-    - torchvision.utils
+    - {{ compiler('cxx') }}
+
+build:
+  number: 0
+  skip: true  # [not linux64]
+
+outputs:
+  - name: torchvision
+    script: build-torch.sh   # [not win]
+    script: build-torch.bat  # [win]
+    build:
+      string: "h{{ PKG_HASH }}_{{ number }}_{{ torch_proc_type }}"                                                  # [cuda_compiler_version == "None"]
+      string: "cuda{{ cuda_compiler_version|replace(".", "") }}h{{ PKG_HASH }}_{{ number }}_{{ torch_proc_type }}"  # [cuda_compiler_version != "None"]
+      run_exports:
+        # ensure matching proc-type
+        - torchvision =*=*_{{ torch_proc_type }}
+
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}         # [cuda_compiler_version != "None"]
+      host:
+        - python
+        - pip
+        - setuptools
+        - cudnn                          # [cuda_compiler_version != "None"]
+        # split off image/video into separate outputs?
+        - jpeg
+        - libpng
+        # ffmpeg
+        - numpy
+        - pillow >=4.1.1
+        - pytorch
+      run:
+        - python
+        - pytorch
```

Comment:
We need to restrict this to `pytorch` cpu builds for `torchvision` cpu builds and `pytorch` gpu builds for `torchvision` gpu builds.